lib/Common/CommonPal.h

```diff
@@ -4,6 +4,12 @@
 //-------------------------------------------------------------------------------------------------------
 #pragma once
 
+// Define _ALWAYSINLINE for template that want to always inline, but doesn't allow inline linkage in clang
+#if defined(__GNUC__) || defined(__clang__)
+#define _ALWAYSINLINE __attribute__((always_inline))
```

Comment:
I suspect, its because, as Curtis' comment says, __forceinline implies inline linkage in some situations- __attribute__((always_inline) does not.